zrml/prediction-markets/src/lib.rs

```diff
@@ -863,13 +893,18 @@ mod pallet {
             let market = T::MarketCommons::market(&market_id)?;
             ensure!(market.status == MarketStatus::Proposed, Error::<T>::InvalidMarketStatus);
             let creator = market.creator;
-            let (imbalance, _) = CurrencyOf::<T>::slash_reserved_named(
+            T::AssetManager::slash_reserved_named(
                 &RESERVE_ID,
+                T::BaseAsset::get(),
                 &creator,
                 T::AdvisoryBond::get(),
             );
-            T::Slash::on_unbalanced(imbalance);
```

Comment:
[This](https://github.com/zeitgeistpm/zeitgeist/blob/main/zrml/prediction-markets/src/lib.rs#L782) is the only occurrence of `T::Slash`. I agree, that we could move the slashed funds to the treasury. I wonder which cases you mean, where the call of `on_unbalanced` is missing?
 
[Here for example? ](https://github.com/zeitgeistpm/zeitgeist/blob/main/zrml/prediction-markets/src/lib.rs#L164-L177). The problem here is, that if the treasury gets the slashed amount then the `admin_destroy_market` would have an incentive to be called, because the `admin` might be the holder of the treasury.

 [Here](https://github.com/zeitgeistpm/zeitgeist/blob/main/zrml/prediction-markets/src/lib.rs#L1533-L1540) is no need to move it to the treasury, because it's moved to the reporter. [Here](https://github.com/zeitgeistpm/zeitgeist/blob/main/zrml/prediction-markets/src/lib.rs#L1575-L1597) it's also moved to the correct reporters.